File: SelfSchoolFE/src/app/classes/material.ts
export class Material {
  private readonly _idMaterial: number = 0;
  private readonly _urlMaterial: string = '';
  private readonly _fileMaterial: string = '';

  constructor(idMaterial: number, urlMaterial: string, fileMaterial: string) {
    this._idMaterial = idMaterial;
    this._urlMaterial = urlMaterial;
    this._fileMaterial = fileMaterial;
  }

  public get idMaterial(): number {
    return this._idMaterial;
  }

  public get urlMaterial(): string {
    return this._urlMaterial;
  }

  public get fileMaterial(): string {
    return this._fileMaterial;
  }
}

File: SelfSchoolBE/SelfSchoolDAL/Entities/Material.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SelfSchoolDAL.Entities
{
    public class Material
    {
        [Key]
        public int idMaterial { get; set; }
        public string urlMaterial { get; set; }
        public string fileMaterial { get; set; }
    }
}
